config/config.go

```diff
@@ -40,21 +42,33 @@ func SetContext(c *cli.Context) error {
 	filename = fname
 
 	// Third, try to open the local VCS repository.
-	_, dir, err := vcs.Nearest(".")
-	if err == vcs.ErrNoNearestVCS {
-		return nil
-	}
+	vcsType, dir, err := vcs.Nearest(".")
 	if err != nil {
-		return err
+		if err == vcs.ErrNoNearestVCS {
+			vcsType = vcs.None
+			dir = "."
+		} else {
+			return err
+		}
 	}
-	r, err := git.PlainOpen(dir)
-	if err == git.ErrRepositoryNotExists {
-		return nil
+
+	var r vcs.System
+
+	switch vcsType {
+	case vcs.Git:
+		r, err = vcs.NewGitRepository(dir)
+	case vcs.Subversion:
+		r, err = vcs.NewSubversionRepository(dir)
+	case vcs.None:
+		r, err = vcs.NewNoRepository(dir)
+	default:
+		err = fmt.Errorf("VCS type %s is not supported", vcsType)
```

Comment:
We should be able to call `errors.Errorf()` but it looks like we haven't provided a wrapper for that in the fossa error's package. This is fine, but you should take a look at`errors/errors.go`.